Python/gc_free_threading.c

```diff
@@ -1758,4 +1758,16 @@ _PyGC_ClearAllFreeLists(PyInterpreterState *interp)
     HEAD_UNLOCK(&_PyRuntime);
 }
 
+void
+_PyGC_Clear_DelayedObjects(PyInterpreterState *interp)
```

Comment:
Let's make this `static` and call it from within `gc_collect_internal` while the world is stopped. A good place might be immediately after merge_all_queued_objects.

If it's static, the naming should reflect that (i.e., lower case). Maybe `process_delayed_frees()`.